src/services/WebService.tsx

```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+import { IPureTodo, ITodo } from '../models/ITodo';
+
+const { REACT_APP_SERVER_URL } = process.env;
+
+class WebService {
+  async getData() {
+    const { data } = await axios.get<ITodo>(REACT_APP_SERVER_URL!);
+    return data;
+  }
+
+  async postData(value: IPureTodo) {
+    const { data } = await axios.post<ITodo>(REACT_APP_SERVER_URL!, value);
+    return data;
+  }
+
+  async deleteData(value: ITodo) {
+    const { data } = await axios.delete<ITodo>(REACT_APP_SERVER_URL!, {
+      data: value,
+    });
+    return data;
+  }
+
+  async updateData(value: ITodo) {
+    await axios.patch(REACT_APP_SERVER_URL!, value);
+  }
+}
+
+export default new WebService();
```

Comment:
Create an instance of the class only where you invoke it.

Reply:
done